v0/destinations/mailchimp/transform.js

```diff
@@ -115,141 +161,169 @@ async function responseBuilderSimple(payload, message, mailChimpConfig) {
+    audienceId,
+    emailExists
+  );
+};
 
-  const mailChimpExists = get(message, "context.MailChimp");
-  if (mailChimpExists) {
-    const listIdExists = get(message, "context.MailChimp.listId");
-    if (listIdExists) {
-      mailChimpConfig.audienceId = message.context.MailChimp.listId;
-    }
+const process = async event => {
+  const { message, destination } = event;
+
+  const destConfig = destination.Config;
+  if (!destConfig.apiKey) {
+    throw new CustomError("API Key not found. Aborting", 400);
   }
-  return mailChimpConfig;
-}
 
-async function processIdentify(message, destination) {
-  const mailChimpConfig = getMailChimpConfig(message, destination);
-  const properties = await getTransformedJSON(message, mailChimpConfig);
-  return responseBuilderSimple(properties, message, mailChimpConfig);
-}
+  if (!destConfig.audienceId) {
+    throw new CustomError("Audience Id not found. Aborting", 400);
+  }
+
+  if (!destConfig.datacenterId) {
+    throw new CustomError("DataCenter Id not found. Aborting", 400);
+  }
+
+  if (!message.type) {
+    throw new CustomError(
+      "Message Type is not present. Aborting message.",
+      400
+    );
+  }
+
+  const messageType = message.type.toLowerCase();
 
-async function processSingleMessage(message, destination) {
-  if (message.type !== EventType.IDENTIFY) {
-    throw new CustomError(`message type ${message.type} is not supported`, 400);
+  let response;
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      response = await identifyResponseBuilder(message, destination);
+      break;
+    default:
+      throw new CustomError(
+        `message type ${messageType} is not supported`,
+        400
+      );
   }
+  return response;
+};
+
+function batchEvents(arrayChunks) {
+  // Batching reference doc: https://mailchimp.com/developer/marketing/api/lists/
+  const batchedResponseList = [];
+
+  // list of chunks [ [..], [..] ]
+  arrayChunks.forEach(chunk => {
+    const batchResponseList = [];
+    const metadata = [];
+
+    // extracting destination
+    // from the first event in a batch
+    const { destination, message } = chunk[0];
+
+    let batchEventResponse = defaultBatchRequestConfig();
 
-  return processIdentify(message, destination);
+    // Batch event into dest batch structure
+    chunk.forEach(ev => {
+      batchResponseList.push(ev.message.body.JSON);
+      metadata.push(ev.metadata);
+    });
+
+    batchEventResponse.batchedRequest.body.JSON = {
+      members: batchResponseList
+    };
+
+    const BASE_URL = `https://${destination.Config.datacenterId}.api.mailchimp.com/3.0/lists/${destination.Config.audienceId}`;
+
+    const BATCH_ENDPOINT = `${BASE_URL}?skip_merge_validation=<false&skip_duplicate_check=true`;
+
+    batchEventResponse.batchedRequest.endpoint = BATCH_ENDPOINT;
+
+    const basicAuth = Buffer.from(
+      `apiKey:${destination.Config.apiKey}`
+    ).toString("base64");
+
+    batchEventResponse.batchedRequest.userId = message.userId
```

Comment:
we can use getFieldValueFromMessage(message, "userId") for this.

Reply:
Done